packages/node/src/runtimePlugin.ts

```diff
@@ -0,0 +1,380 @@
+        'for',
+        chunkName,
+        'public path:',
+        __webpack_require__.p,
+      );
+      url = new URL(remotes[remoteName]);
+      var getBasenameFromUrl = function (url) {
+        var urlParts = url.split('/');
+        return urlParts[urlParts.length - 1];
+      };
+      var fileToReplace = getBasenameFromUrl(url.pathname);
+      url.pathname = url.pathname.replace(fileToReplace, chunkName);
+    }
+    fetch(url)
+      .then(function (res) {
+        return res.text();
+      })
+      .then(function (data) {
+        var chunk = {};
+        try {
+          var urlDirname = url.pathname.split('/').slice(0, -1).join('/');
+          eval(
+            '(function(exports, require, __dirname, __filename) {' +
+              data +
+              '\n})',
+          )(chunk, require, urlDirname, chunkName);
+          callback(null, chunk);
+        } catch (e) {
+          callback(e, null);
+        }
+      });
+  }
+
+  function httpVmStrategy(chunkName, remoteName, remotes, callback) {
+    var http = require('http');
+    var https = require('https');
+    var vm = require('vm');
+    var path = require('path');
+    var url;
+
+    var globalThisVal = new Function('return globalThis')();
+    try {
+      url = new URL(chunkName, __webpack_require__.p);
+    } catch (e) {
+      console.error(
+        'module-federation: failed to construct absolute chunk path of',
+        remoteName,
+        'for',
+        chunkName,
+        'for public path',
+        __webpack_require__.p,
+      );
+      // search all instances to see if any have the remote
+
+      var entryUrl;
+      var container = globalThisVal['__FEDERATION__']['__INSTANCES__'].find(
+        function (instance) {
+          if (!instance.moduleCache.has(remoteName)) return null;
+          var container = instance.moduleCache.get(remoteName);
+          if (!container.remoteInfo) return null;
+          return container;
+        },
+      );
+
+      if (container) {
+        entryUrl = container.moduleCache.get(remoteName).remoteInfo.entry;
+      } else {
+        var currentName = __webpack_require__.federation.initOptions.name;
+        var backupContainer = __FEDERATION__.__INSTANCES__.find(
+          function (hostInstance) {
+            return hostInstance.options.remotes.find(function (remote) {
+              if (remote.name === currentName || remote.alias === currentName) {
+                return true;
+              }
+              return false;
+            });
+          },
+        );
+
+        if (backupContainer) {
+          var backupRemote = backupContainer.options.remotes.find(
+            function (remote) {
+              return (
+                remote.name === currentName || remote.alias === currentName
+              );
+            },
+          );
+          entryUrl = backupRemote.entry;
+        }
+      }
+
+      if (!entryUrl) {
+        var mockChunk = { modules: {}, ids: [], runtime: false };
```

Comment:
If chunk handler cannot obtain resource then it is most likely not a chunkId intended to be handled by js chunk handler (like png chunk, internal MF chunk)

To prevent chunk load callback from stalling, i return callback for chunk handler upon no possible load solution and hope that another handler will catch the chunk.